src/Microsoft.Performance.SDK/Processing/DataSource.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.Performance.SDK.Processing
+{
+    /// <summary>
+    ///     Base of all Data Source implementations.
+    /// </summary>
+    public abstract class DataSource
+        : IDataSource
+    {
+        private readonly Uri uri;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="DataSource"/>
+        ///     class with the given <see cref="Uri"/>.
+        /// </summary>
+        /// <param name="uri">
+        ///     The URI of the data.
+        /// </param>
+        /// <exception cref="System.ArgumentNullException">
+        ///     <paramref name="uri"/> is <c>null</c>.
+        /// </exception>
+        protected DataSource(Uri uri)
+        {
+            Guard.NotNull(uri, nameof(uri));
+
+            this.uri = uri;
+        }
+
+        /// <summary>
+        ///     Returns the <see cref="Uri" /> of the data.
+        /// </summary>
+        /// <returns>
+        ///     The URI of the data.
+        /// </returns>
+        public Uri GetUri()
```

Comment:
while we're here, why not make this a readonly property

Reply:
Sure